test-case/verify-ucm-config.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+##
+## Case Name: verify-ucm-config
+## Preconditions:
+##    N/A
+## Description:
+##    verify the ucm configuration
+## Case step:
+##    1. load the ucm config file
+##    2. check ucm can be run by alsaucm correctly or not:
+##       check the ucm configuration file meets the syntax or not
+##       check if the devices can be enabled or not
+##       check if the controls can be set correctly or not
+##       and so on. Details can be found in cmd 'alsaucm --help'
+## Expect result:
+##    command line check with $? without error
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+func_opt_parse_option $*
+
+func_verify_verb_device()
+{
+    local cardname=$1
+    verb=$2
+    # get all the devices used by the verb
+    dlogc "alsaucm -c '$cardname' set _verb $verb list _devices"
+    devices=$(alsaucm -c "$cardname" set _verb $verb list _devices)
+    [[ $? -ne 0 ]] && dloge "list device failed" && return 1
+    ((descline=0))
+    local OLD_IFS="$IFS"
+    IFS=$'\n'
+    for device in $devices; do
+        if [ $descline -eq "0" ]; then
+            device=`echo $device |awk -F ": " '{print $2}'`
+            [[ -z "$verb" ]] && dlogw "device is null, skipping" && continue
+            dlogi "found device: $device"
+
+	    # enable the device and check if the device is really enabled or not
+            dlogc "alsaucm -c '$cardname' set _verb '$verb' set _enadev '$device' list1 _enadevs"
+            ret=`alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" list1 _enadevs`
+            [[ $? -ne 0 ]] && dloge "enable '$verb':'$device' failed" && IFS="$OLD_IFS" && return 1
+            ret=`echo $ret |grep "$device" 2>/dev/null`
+            [[ -z "$ret" ]] && dloge "'$verb':'$device' is not in the enabled device list" && IFS="$OLD_IFS" && return 1
+
+	    # enable the device and setup the controls to check if the controls can be set correctly or not
+            dlogc "alsaucm -c '$cardname' set _verb '$verb' set _enadev '$device' reload"
+            alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" reload
+            [[ $? -ne 0 ]] && dloge "reload '$verb':'$device' setting failed!" && IFS="$OLD_IFS" && return 1
+
+            # device format: the first line is device and the second line is description
+            # Let's skip the description
+            ((descline=1))
+        else
+            ((descline=0))
+        fi
+    done
+    IFS="$OLD_IFS"
+    return 0
+}
+
+ucm_folder=/usr/share/alsa/ucm
+
+sofcard=${SOFCARD:-0}
+cardname=$(sof-dump-status.py -s $sofcard)
+[[ $? -ne 0 ]] && exit 1
+
+# TODO: check the audio card dmi name (dmi name is not recommanded).
+ucm_file=${ucm_folder}/${cardname}/${cardname}.conf
```

Comment:
this case need libasound2-data package to install/some thing, I think you need update other file to add this package

Reply:
we need updated libasound2.so. This is another feature. I'm doing it in another thread.